

Updated version of an older patch against 1.16.x.  Illumos has FIONREAD
in <sys/filio.h>


diff -ur gst-plugins-base-1.18.5.orig/gst/tcp/gstmultifdsink.c gst-plugins-base-1.18.5/gst/tcp/gstmultifdsink.c
--- gst-plugins-base-1.18.5.orig/gst/tcp/gstmultifdsink.c	2021-09-08 14:02:22.195757900 +0000
+++ gst-plugins-base-1.18.5/gst/tcp/gstmultifdsink.c	2021-10-25 21:49:33.517466957 +0000
@@ -119,6 +119,11 @@
 #include <sys/stat.h>
 #include <netinet/in.h>
 
+#if defined(__sun) && defined(__SVR4)
+/* for FIONREAD */
+#include <sys/filio.h>
+#endif
+
 #include "gstmultifdsink.h"
 
 #define NOT_IMPLEMENTED 0
